tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -451,9 +456,48 @@ func (a *AwsProvider) findOrCreateSg(infraID string, vpc *ec2.Vpc) (string, erro
 		log.Info(fmt.Sprintf("Winrm port already opened for Security Group: %s.", sgID))
 	}
 
+	// Check ssh port is open
+	isSSHPortOpen, err := a.checkSgSSHPortIsOpen(sgID)
+	if err != nil {
+		return "", err
+	}
+
+	// Add ssh port to security group if it doesn't exist
+	if !isSSHPortOpen {
+		err := a.addSSHPortToSG(sgID)
+		if err != nil {
+			return "", err
+		}
+		log.Info(fmt.Sprintf("ssh port is now added to Security Group %s", sgID))
+	} else {
+		log.Info(fmt.Sprintf("ssh port already opened for Security Group: %s.", sgID))
+	}
+
 	return sgID, nil
 }
 
+// checkSgSSHPortIsOpen checks whether the ssh port is opened in the given security group.
+func (a *AwsProvider) checkSgSSHPortIsOpen(sgId string) (bool, error) {
```

Comment:
Please rename to `checkSSHPortIsOpenInSG` and a TODO to make this a common helper for checking if ports are open in SGs.